### ISSUE TITLE
Youtube doesn't display thumbnails videos !

### ISSUE THREAD

**Issue by floalma** [dillbyrne/random-agent-spoofer#306]
Hi,

This is to notify that it doesn't work on youtube, when clicking on user's videos listing, I can't view the thumbnails, even with with all the 'Standard Options' disabled and also it' doesn't work if I check 'Real Profile'. Please fix it soon.

Thanks in advance
Floalma

**Comment by sharan01** [comment]
if user agent is set ot chrome/chromium then youtube loads webp images which firefox does not understand
whitelist youtube in the settings or change useragent to different one

**Comment by floalma** [comment]
Sharan,

1.Is this only with Chrome browser ?  How can I choose all the others browsers except of course Chrome builds or to choose on random all the firefox builds ? Is it possible ? 

2. Request: it will great to add fonction to disable RAS when clicking on the RAS icon, instead to do it on the Add-ons display list (firefox).

3. I also noticed that when I click on the 'Real Profile', it doesn't really work, not take effect on the browser.  Could you please check it and fix it ?

4. On gmail, clicking on 'New Message' with RAS, nothing is happening, not seems to work.

Thanks in advance.

**Comment by sharan01** [comment]
I am not the author of this addon
 you can exclude chrome browsers from random selecting 
in the profile tab
![screenshot](https://cloud.githubusercontent.com/assets/1240154/10715238/ef12a8c8-7b34-11e5-9c6a-caca387761e3.png)

**Comment by floalma** [comment]
Great Sharan, I didn't see the exclude button. Thank you.

Do you have an idea about the question 2 and 3, please ?

**Comment by RoxKilly** [comment]
For question 3, try `RAS icon -> Options tab` and uncheck `Disable Canvas`.  If that works, you could whitelist the canvas element for the gmail domain:
*  go to `RAS icon ->  Whitelist tab -> Whitelist Rules` and between the square brackets, add `{"url": "example.com", "options": ["canvas"]}`, substituting the real gmail domain for `example.com`.  If there is already something inside the square braces, separate it from your new addition with a comma `,`.   On that same tab (near the top), make sure `Enable whitelist` is checked.

**Comment by floalma** [comment]
Thanks RoxKilly, it works for question 3, nevertheless, I can't reply any message and I can't wrote a message with your customization. Do you know how to do it ?

Do you experimented the same for the question 2 ?

**Comment by RoxKilly** [comment]
For question 3: I'm not sure which options will get gmail to fully work.  If things work when you select `RAS -> Profile tab -> Real Profile` you'll need to test things to find out what works. Try `RAS icon -> Options tab` and uncheck `Enable script injection` near the top. If that doesn't work, you could also try unchecking the protections in the `Headers tab` one by one until you find out what works.

For question 2 I'm not sure what you mean by "it doesn't take effect on the browser" What do you expect to happen, and what's actually happening?

**Comment by floalma** [comment]
For Answer 3: It works, thank you RoxKilly.

About question 2: I mean that if I choose a 'Real Profile', I'm not surfing as a 'Real Profile' but with the customization options and settings. Real Profile, doesn't work for me

Another question: If I disable RAS, then I enable again, I lost all my settings. Have you experienced that ?

**Comment by RoxKilly** [comment]
>  if I choose a 'Real Profile', I'm not surfing as a 'Real Profile' but with the customization options and settings. Real Profile, doesn't work for me

How can you tell that the customized options and settings are still in effect? As for RAS losing its settings when you disable it, this is a known limitation unfortunately. See issues #270 and #273

**Comment by dillbyrne** [comment]
The whitelist purposely has youtube entries specifically for this purpose.

@floalma which version of RAS are you using ? you can find this on the right most tab labelled `?`

**Comment by floalma** [comment]
@RoxKilly 
"As for RAS losing its settings when you disable it, this is a known limitation unfortunately"
Yes, it wil be great to add this settings.
I know that 'Real Profile' doesn't work, because the display doesn't change if I compare with my customize settings. The display is the same, not different for those sites that RAS can't display it properly.

@dillbyrne   My version is 0.9.5.3

**Comment by floalma** [comment]
@dillbyrne 
Any news ?

**Comment by dillbyrne** [comment]
@floalma With the default whitelist on youtube for 0.9.5.3 it should display the thumbnails.

![ta](https://cloud.githubusercontent.com/assets/2903711/11102488/c118629a-88b4-11e5-96e4-070e117ba00a.png)

This is a fresh install of RAS with all options checked. The whitelist profile is the default in 0.9.5.3 (ff 37) and the whitelist rules are also the default which include youtube.

```
[{"url": "play.google.com"}, 
{"url": "s.ytimg.com", "options": ["canvas"]}, 
{"url": "youtube.com", "options": ["canvas"]}]
```

![erw](https://cloud.githubusercontent.com/assets/2903711/11102581/4bc97a32-88b5-11e5-99dd-43afcb3e5554.png)


The s.ytimg.com rule is for youtube images 

Script injection is required for whitelisting to work

**Comment by ipatjolvur** [comment]
@floalma, each of your three questions should be in separate issue posts.

About Youtube's thumbnails, I am using RAS 0.9.5.5, and all Chrome and Chromium profiles block Youtube's thumbnails. In the default whitelist in 0.9.5.5, the rule, `{"url":"youtube.com"}`, allows the thumbnails. The older rule had `"options": ["canvas"]`, but it is not needed anymore.  The other old rule, `{"url": "s.ytimg.com", "options": ["canvas"]}`, does not affect them. The webp images are stored as of now on i.ytimg.com, but adding a rule for `{"url":"i.ytimg.com"}`, even with all of the options, does not affect the thumbnails.

In summary, enabling the default whitelist fixed it, and sending the whitelist's profile is a better solution than excluding the Chrome and Chromium profiles from all sites. Thank you.

**Comment by dillbyrne** [comment]
@ipatjolvur Thank you :+1: 

As of ff 45 script injection is broken and the whitelist depends on script injection. I am slowly rewriting the script injection code to make use frame scripts. So if you are using ff45 the whitelist will not work until I release a new version with the necessary fixes.
